### PR TITLE
feat(medusa-payment-stripe): Enhance stripe payment to handle description and improve status resolution

### PR DESCRIPTION
**Pull request by adrien2p**
Allow the user to either set a static description through configuration such as:

```js
  {
    resolve: `medusa-payment-stripe`,
    options: {
      api_key: STRIPE_API_KEY,
      webhook_secret: STRIPE_WEBHOOK_SECRET,
      payment_description: "Digital arts/services",
    },
  },
```

Or eventually override the static description through the cart metadata.

### CODE REVIEW

**Comment by adrien2p** [comment]
Fixes https://github.com/medusajs/medusa/issues/1401

**Comment by adrien2p** [comment]
@srindom the problem is that you need to create the payment intent with the description, otherwise you get an error and therefore there is nothing to update ☺️

**Comment by ShivamJoker** [comment]
Ah! But still wouldn't it be easier to have a global config at least instead of updating it on cart?

**Comment by adrien2p** [comment]
@srindom any suggestion regarding the doc I linked in my comment?

**Comment by olivermrbl** [comment]
Given that descriptions are required on payments in some countries, I would argue that adding it as a configurable prop to the plugin options is reasonable. 

You wrote "& other params" in the feature description @ShivamJoker - do you know of any other properties that might be worth looking into before we proceed with this one?

**Comment by ShivamJoker** [comment]
I thought for other countries we may need some more params, so I thought but for now this would do the job.

File: packages/medusa-payment-stripe/src/services/stripe-provider.js [hunk 1]
```diff
@@ -141,6 +125,7 @@ class StripeProviderService extends PaymentService {
     const amount = await this.totalsService_.getTotal(cart)
 
     const intentRequest = {
+      description: cart?.context?.description ?? this.options?.description,
```
Comment (olivermrbl) [inline packages/medusa-payment-stripe/src/services/stripe-provider.js:128]: Would argue, that plugin options should take precedence over cart context

**Review by olivermrbl — COMMENTED** [review]
Should we change the plugin property to `payment_description` to make it clear, what is actually configured?

**Comment by ShivamJoker** [comment]
I think yeah we can do that.

File: packages/medusa-payment-stripe/src/services/stripe-provider.js (hunk 1 above)
Comment (adrien2p) [reply packages/medusa-payment-stripe/src/services/stripe-provider.js:128]: I wasn't sure, i was thinking that if you want to out a special description depending on the provenance of the order etc if the configuration is taking precedence it means that you can't. Wdty? The description could integrate some interesting information about the context of the payment

**Comment by adrien2p** [comment]
> Should we change the plugin property to `payment_description` to make it clear, what is actually configured?

Yeah agree 👍 ill do the changes asap

**Comment by adrien2p** [comment]
@olivermrbl could you point me to where I should update the doc. Only if you want it to be part of the pr

File: packages/medusa-payment-stripe/src/services/stripe-provider.js (hunk 1 above)
Comment (adrien2p) [reply packages/medusa-payment-stripe/src/services/stripe-provider.js:128]: @olivermrbl wdty?

**Comment by olivermrbl** [comment]
@adrien2p did you test, that an `undefined` description is allowed when creating the payment intent?

**Pull request merged**